docs/birthday-kata.md

```diff
@@ -0,0 +1,385 @@
+
+In our case, let's limit ourselves to the following effects: sending a message, fetching employees from somewhere (a flatfile, a db, an api, anything), and fetching the current date. For the sake of simplicity let's ignore an exception throwing ability (and failures in general) or any other abilities that may make sense to add in a real production system.
+
+```unison
+ability MessageService where
+  send: Message ->{MessageService} Message
+
+ability EmployeeRepository where
+  fetchAllByBirthday: (Day, Month) ->{EmployeeRepository} [Employee]
+
+ability Calendar where
+  today: '{Calendar} Date
+```
+
+Abilities can sort of be thought of as providing interfaces, and interfaces in turn can be thought of as providing commands.
+
+Let's walk through what these abilities are doing:
+1) `MessageService` is an ability that provides the interface for a single command `send`. `send`'s type signature can be read as "a function that takes a message and returns that same message, requiring the `MessageService` ability. The compiler is now aware of side effects in the type signature of functions!
+2) `EmployeeRepository` provides the interface `fetchAllByBirthday` which takes a `Day` and a `Month` pair and returns a list of `Employees`, requiring the `EmployeeRepository` ability.
+3) `Calendar` provides the interface `today` which is a function that takes no arguments and returns a `Date`, requiring the `Calendar` ability.
+
+Note that abilities 
+1) can provide multiple interfaces if we wanted
+```unison
+ability EmployeeRepository where
+  fetchAllByBirthday: Day -> Month ->{EmployeeRepository} [Employee]
+  fetchAll: '{EmployeeRepository} [Employee]
+  findById: EmployeeId -> {EmployeeRepository} {Employee}
+```
+2) have there behavior defined elsewhere (using handler syntax we will get to later)
+3) are composable (hence "algebraic effects")
+4) require that any calling function provide the ability in their type signature, which bubbles all the way to the top of the call stack (until it finds a handler).
+
+So let's scaffold our main function for this exercise and see this stuff in action.
+
+```unison
+sendBirthdayEmails : '[Message]
+sendBirthdayEmails = 
+  todo "get the current date"
+  todo "fetch all the employees with birthdays today"
+  todo "create happy birthday messages"
+  todo "send and return the messages"
+```
+Let's say we want a function we can call that will find all the employees whose birthday is today, create emails for them, send those emails, and then return those emails so we can see what was created and sent.
+
+As an aside, let's avoid modeling failures simply because it will clutter this exercise a bit. In real life, many of our effects would have type signatures with `Either` so we could model possible failures, but let's just naively pretend everything always works.
+
+So let's do some more scaffolding!
+```unison
+sendBirthdayEmails : '[Message]
+sendBirthdayEmails = 
+  today' = today
```

Comment:
does `today` automatically get called here? should it be `!today` since otherwise its a deferred computation by default?